content/en/blog/_posts/2023-12-05-kubernetes-1.29.md

```diff
@@ -0,0 +1,183 @@
+
+### Node volume expansion Secret support for CSI drivers ([sig-storage](https://github.com/kubernetes/community/tree/master/sig-storage))
+    
+In Kubernetes, a volume expansion operation may include the expansion of the volume on the node, which involves filesystem resize. Some CSI drivers require secrets, for example a credential for accessing a SAN fabric, during the node expansion for the following use cases:
+	•	When a PersistentVolume represents encrypted block storage, for example using LUKS, you need to provide a passphrase in order to expand the device.
+	•	For various validations, the CSI driver needs to have credentials to communicate with the backend storage system at time of node expansion. 
+
+To meet this requirement, a CSI Node Expand Secret feature was introduced in Kubernetes 1.25. This allows an optional secret field to be sent as part of the NodeExpandVolumeRequest by the CSI drivers so that node volume expansion operation can be performed with the underlying storage system. In Kubernetes 1.29, this feature became generally available.
+
+## What's New in Beta
+
+### Per-plugin callback functions in kube-scheduler for accurate requeueing ([sig-scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+The throughput of the scheduler is our eternal challenge. This QueueingHint feature brings a new possibility to optimize the efficiency of requeueing, which could reduce useless scheduling retries significantly.
+
+### Node lifecycle separated from taint management ([sig-scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling))
+
+As title describes, it's to decouple `TaintManager` that performs taint-based pod eviction from `NodeLifecycleController` and make them two separate controllers: `NodeLifecycleController` to add taints to unhealthy nodes and `TaintManager` to perform pod deletion on nodes tainted with NoExecute effect.
+
+## What's New in Alpha
+
+### Define Pod affinity or anti-affinity using `matchLabelKeys` ([sig-scheduling](https://github.com/kubernetes/community/tree/master/sig-scheduling)) 
+
+The throughput of the scheduler is our eternal challenge. This QueueingHint feature brings a new possibility to optimize the efficiency of requeueing, which could reduce useless scheduling retries significantly. We're actively working on this, and some stable in-tree plugins will have supported it in v1.29.
+
+### nftables backend for kube-proxy ([sig-network](https://github.com/kubernetes/community/tree/master/sig-network))
+    
+The default kube-proxy implementation on Linux is currently based on iptables. IPTables was the preferred packet filtering and processing system in the Linux kernel for many years (starting with the 2.4 kernel in 2001). However, unsolvable problems with iptables led to the development of a successor, nftables. Development on iptables has mostly stopped, with new features and performance improvements primarily going into nftables instead.
```

Comment:
```suggestion
The default kube-proxy implementation on Linux is currently based on iptables. Iptables was the preferred packet filtering and processing system in the Linux kernel for many years (starting with the 2.4 kernel in 2001). However, unsolvable problems with iptables led to the development of a successor, nftables. Development on iptables has mostly stopped, with new features and performance improvements primarily going into nftables instead.
```

Reply:
Nope. It's iptables, all lowercase. We'd consider rephrasing to avoid the capitalization question at the start of a sentence, similar to what we do in the docs.

Reply:
Eg:
> For many years, iptables was the preferred packet filtering and processing system in the Linux kernel (going back to with the 2.4 kernel released in 2001).

Reply:
> 

```suggestion
The default kube-proxy implementation on Linux is currently based on iptables. This was the preferred packet filtering and processing system in the Linux kernel for many years (starting with the 2.4 kernel in 2001). However, unsolvable problems with iptables led to the development of a successor, nftables. Development on iptables has mostly stopped, with new features and performance improvements primarily going into nftables instead.
```